### PR TITLE
Made several wording adjustments.

### PR DESCRIPTION
**Pull request by camio**
I tried to make the text flow a bit better and to soften some of the
rough edges.

### CODE REVIEW

File: development/submissions.html [hunk 1]
```diff
@@ -139,34 +140,33 @@
 
               <p>A message to the Boost developers mailing list
               might be as simple as "Is there any interest in a
-              library which solves Travelling Salesperson problems in linear
-              time?"</p>
+              library which solves X?"</p>
```
Comment (ned14) [inline development/submissions.html:143]: I think a realistic example rather than 'X' would be valuable

File: development/submissions.html [hunk 2]
```diff
@@ -264,39 +264,38 @@
...
+              response to feedback and reviews. For bigger ticket items of
+              work, open issues on your issue tracker so interested people can
               track the fixing of specific issues raised.</p>
 
               <p>The review manager will consider all the reviews made by
               members of the community and arrive at a decision on
               whether your library is rejected, conditionally accepted or
-              unconditionally accepted. They will post a report summarising
-              their decision publicly. If conditions are attached to
+              unconditionally accepted. He or she will post a report summarising
```
Comment (ned14) [inline development/submissions.html:297]: I'd be careful here, some people identify as non-gendered. "They" avoids problems.

File: development/submissions.html [hunk 3]
```diff
@@ -331,8 +330,9 @@
               mailing lists can alert you to potential maintenance needs;
               please plan to maintain your library over time. If you no
               longer can or wish to maintain your library, please post a
-              message on the Boost developers mailing list and help someone
-              else take over as the library maintainer.</p>
+              message on the Boost developers mailing list asking for a new
+              maintainer to volunteer and then spend the time to help him or
```
Comment (ned14) [inline development/submissions.html:334]: Same goes here with fixed gender specification

File: development/submissions.html (hunk 1 above)
Comment (camio) [reply development/submissions.html:143]: Sure, I don't feel strongly about this.

File: development/submissions.html (hunk 2 above)
Comment (camio) [reply development/submissions.html:297]: What about the people who don't identify as plural? Feel free to do whatever you think is best here.

**Pull request merged**